### PR TITLE
Get module components from corgi and save the components

### PR DESCRIPTION
**Pull request by huiwangredhat**
[OLCS-251](https://issues.redhat.com/browse/OLCS-251)
Fixes OLCS-251

### CODE REVIEW

File: openlcs/libs/corgi_handler.py [hunk 1]
```diff
@@ -198,3 +198,59 @@ def get_product_version(self, fields=None) -> dict:
...
+
+    def get_components_data(self):
+        """
+        Get the RHEL_MODULE components
+        """
+        executor = ThreadPoolExecutor(max_workers=5,)
+        event_loop = asyncio.get_event_loop()
+        components = []
+        component_links, module_component = self.get_component_and_links(
+            self.module_nvr)
```
[the remark this answers is not in the document]
Comment (huiwangredhat) [reply openlcs/libs/corgi_handler.py:245]: Updated. Thanks.

File: openlcs/packages/mixins.py [hunk 2]
```diff
@@ -363,3 +363,59 @@ def save_container_components(self, **kwargs):
             self.build_release_node(container_component)
         else:
             self.build_container_node(container_component)
+
+
+class SaveModuleComponentsMixin(SaveContainerComponentsMixin):
```
[the remark this answers is not in the document]
Comment (huiwangredhat) [reply openlcs/packages/mixins.py:368]: Good point. I think it's more clear for debugging the module and debugging the container separately. You know , now container functions are not very stable, and we still need time to fix the existing issues.  If the two parts are stable, let's move them together later.

File: openlcs/libs/corgi_handler.py [hunk 3]
```diff
@@ -198,3 +198,59 @@ def get_product_version(self, fields=None) -> dict:
             for field in fields:
                 retval[field] = product_data[field]
         return retval
+
+
+class ModuleComponentsAsync(ContainerComponentsAsync):
```
[the remark this answers is not in the document]
Comment (huiwangredhat) [reply openlcs/libs/corgi_handler.py:203]: Good point. Same reply as the one below.

**Comment by hanchuntao** [comment]
LGTM

File: openlcs/packages/mixins.py [hunk 4]
```diff
@@ -363,3 +363,59 @@ def save_container_components(self, **kwargs):
...
+                    content_type=component_ctype,
+                    object_id=component.id,
+                )
+
+    def save_module_components(self, **kwargs):
+        self.components = kwargs.get('components')
+        product_release = kwargs.get('product_release')
+
+        # Create module parent components
+        print(self.components)
```
[the remark this answers is not in the document]
Comment (huiwangredhat) [reply openlcs/packages/mixins.py]: Thanks. Deleted it.

File: openlcs/packages/mixins.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (huiwangredhat) [reply openlcs/packages/mixins.py]: Updated with your comment. Pls, help to review it again. Thanks.

File: openlcsd/flow/tasks.py [hunk 5]
```diff
@@ -928,6 +931,49 @@ def save_container_components(context, engine):
...
+    """
+    config = context.get('config')
+    engine.logger.info("Start to get module components data...")
+    mc = ModuleComponentsAsync(
+        config.get('CORGI_API_PROD'), context.get('package_nvr'))
+    context['components'] = mc.get_components_data()
+    engine.logger.info("Finished getting container components data.")
+
+
+def save_module_components(context, engine):
```
[the remark this answers is not in the document]
Comment (huiwangredhat) [reply openlcsd/flow/tasks.py]: Updated.

File: openlcs/libs/corgi_handler.py [hunk 6]
```diff
@@ -148,6 +150,9 @@ async def get_event_loop(self, executor, component_links):
         """
         Event loop function for component links.
         """
+        # Fix the the issue RuntimeError: Event loop is closed
+        loop = asyncio.new_event_loop()
+        asyncio.set_event_loop(asyncio.new_event_loop())
```
[the remark this answers is not in the document]
Comment (huiwangredhat) [reply openlcs/libs/corgi_handler.py]: Updated.

File: openlcsd/flow/tasks.py [hunk 7]
```diff
@@ -897,35 +897,52 @@ def get_container_remote_source(context, engine):
...
+    else:
+        component_type = 'RHEL_MODULE'
     data = {
         'components': context.get('components'),
-        'product_release': context.get('product_release')
+        'product_release': context.get('product_release'),
+        'component_type': component_type
     }
-    msg = f'Start to save components in container {package_nvr}...'
+    msg = f'Start to save components in container/module {package_nvr}...'
```
[the remark this answers is not in the document]
Comment (huiwangredhat) [reply openlcsd/flow/tasks.py]: Updated.

**Pull request merged**